### PR TITLE
NodeJS bindings for Tiler

### PR DESCRIPTION
**Pull request by pierotofy**
Adds bindings for tiler code, unit tests, fixes, system command.

### CODE REVIEW

File: src/thumbs.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@
 namespace ddb{
 
 fs::path getThumbFromUserCache(const fs::path &imagePath, time_t modifiedTime, int thumbSize, bool forceRecreate){
-    if (std::rand() % 100 == 0) cleanupThumbsUserCache();
+    if (std::rand() % 1000 == 0) cleanupThumbsUserCache();
```
Comment (HeDo88TH) [inline src/thumbs.cpp:20]: How is this number calculated?

File: src/tiler.cpp [hunk 2]
```diff
@@ -339,12 +341,49 @@ BoundingBox<int> TilerHelper::parseZRange(const std::string &zRange){
...
-fs::path TilerHelper::getCacheFolderName(const fs::path &geotiffPath, time_t modifiedTime){
+fs::path TilerHelper::getCacheFolderName(const fs::path &geotiffPath, time_t modifiedTime, int tileSize){
     std::ostringstream os;
-    os << geotiffPath.string() << "*" << modifiedTime;
+    os << geotiffPath.string() << "*" << modifiedTime << "*" << tileSize;
     return Hash::strCRC64(os.str());
 }
 
+fs::path TilerHelper::getFromUserCache(const fs::path &geotiffPath, int tz, int tx, int ty, int tileSize, bool tms, bool forceRecreate){
+    if (std::rand() % 1000 == 0) cleanupUserCache();
```
Comment (HeDo88TH) [inline src/tiler.cpp:351]: How is this number calculated?

**Comment by HeDo88TH** [comment]
Only windows CI is failing with this
`error MSB6001: Invalid command line switch for "cmd.exe". System.ArgumentException: Item has already been added. Key in dictionary: 'NPM_CONFIG_CACHE'  Key being added: 'npm_config_cache' [D:\a\DroneDB\DroneDB\build\ZERO_CHECK.vcxproj`
Should we merge anyways?

File: src/tiler.cpp (hunk 2 above)
Comment (pierotofy) [reply src/tiler.cpp:351]: Arbitrary. We need to cleanup the cache once in a while without user intervention. Not sure what would be a better strategy. A counter would require shared access to files.

**Comment by pierotofy** [comment]
I'll look into the NodeJS CI error.

File: src/thumbs.cpp (hunk 1 above)
Comment (pierotofy) [reply src/thumbs.cpp:20]: Same (arbitrary). Suggestions to improve are welcome.

**Pull request merged**